### PR TITLE
RWA-1718-Fix - Move health check into its own rest controller

### PR DESCRIPTION
**Pull request by sullivangareth**
### Change description ###

 Move health check into its own rest controller so that it doesn't affect PR pipelines

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by serdaroguzhanakin — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthEndpoint.java [hunk 1]
```diff
@@ -28,12 +35,13 @@ public class ReceivedMessagesHealthEndpoint implements HealthIndicator {
...
-        return HealthIndicator.super.getHealth(includeDetails);
-    }
-
-    @Override
+    @Operation(description = "Query case event message db to find problematic messages by job name")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "OK", content = {
+            @Content(mediaType = "application/json", schema = @Schema(implementation = Health.class))})
+    })
+    @GetMapping(path = "/ccdMessagesReceived/health")
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthEndpoint.java:43]: I would probably name this path differently

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthEndpoint.java [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
+import org.springframework.web.bind.annotation.RestController;
 import uk.gov.hmcts.reform.wacaseeventhandler.repository.CaseEventMessageRepository;
 import uk.gov.hmcts.reform.wacaseeventhandler.services.holidaydates.HolidayService;
 
 import java.time.Clock;
 import java.time.LocalDateTime;
 
-@Component("ccdMessagesReceived")
-public class ReceivedMessagesHealthEndpoint implements HealthIndicator {
+@RestController
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthEndpoint.java:20]: I think for the immediate problem of PRs failing we can simply stop this from being included in as a HealthIndicator

**Review by raghera — CHANGES_REQUESTED** [review]
As discussed it is worth thinking this through a bit first.  There's the immediate issue of PRs not building which I think is probably easiest to address by excluding this as a HealthIndicator.  

Then we have to think of a pattern that works and consider all the different envs.

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthEndpoint.java (hunk 2 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthEndpoint.java:20]: This change does exclude it from the HealthIndicator

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthEndpoint.java (hunk 1 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthEndpoint.java:43]: 👍

**Comment by rafalkalita** [comment]
ccdMessagesReceived is using now management.endpoint.health.receivedMessageCheckEnvEnabled property to indicate where to execute.

**Pull request closed without merging**